dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -113,3 +203,24 @@ def _show_selected_keys_from_dict(dct, keys: list):
         for key in keys:
             dct_selected.append(dct[key])
         return dict(zip(keys, dct_selected))
+
+    def _remap_dimension_columns(self, df):
+        # This will likely become a common activity when running queries.
+        # May need to cache the result. But that will likely be in Project, not here.
+        # This method could be moved elsewhere.
+        for ref in self._mapping_references:
+            column = ref.from_dimension_type.value
+            if column in df.columns:
+                mapping_config = self._dimension_mapping_mgr.get_by_id(
+                    ref.mapping_id, version=ref.version
+                )
+                records = models_to_dataframe(mapping_config.model.records).filter(
+                    "to_id is not NULL"
+                )
+                df = (
+                    df.join(records, df[column] == records.from_id)
+                    .drop("from_id")
+                    .drop(column)
+                    .withColumnRenamed("to_id", column)
+                )
```

Comment:
We need to change how dimension columns get remapped to handle duplicates in from_id.

Reply:
To be addressed in #108